src/definitions/device_engagement.rs

```diff
@@ -356,59 +410,67 @@ impl From<BleOptions> for CborValue {
                 uuid,
                 ble_device_address,
             }) => {
-                map.insert(CborValue::Integer(0), CborValue::Bool(true));
-                map.insert(
-                    CborValue::Integer(10),
-                    CborValue::Bytes(uuid.as_bytes().to_vec()),
-                );
+                map.push((
+                    ciborium::Value::Integer(0.into()),
+                    ciborium::Value::Bool(true),
+                ));
+                map.push((
+                    ciborium::Value::Integer(10.into()),
+                    ciborium::Value::Bytes(uuid.as_bytes().to_vec()),
+                ));
                 if let Some(address) = ble_device_address {
-                    map.insert(CborValue::Integer(20), address.into());
+                    map.push((
+                        ciborium::Value::Integer(20.into()),
+                        cbor::into_value(address).unwrap(),
+                    ));
                 }
             }
             None => {
-                map.insert(CborValue::Integer(0), CborValue::Bool(false));
+                map.push((
+                    ciborium::Value::Integer(0.into()),
+                    ciborium::Value::Bool(false),
+                ));
             }
         }
 
-        CborValue::Map(map)
+        ciborium::Value::Map(map)
     }
 }
 
-impl TryFrom<CborValue> for WifiOptions {
+impl TryFrom<ciborium::Value> for WifiOptions {
     type Error = Error;
 
-    fn try_from(v: CborValue) -> Result<Self, Error> {
+    fn try_from(v: ciborium::Value) -> Result<Self, Error> {
         fn lookup_opt_string(
-            map: &BTreeMap<CborValue, CborValue>,
+            map: &BTreeMap<i128, ciborium::Value>,
             idx: i128,
         ) -> Result<Option<String>, Error> {
-            match map.get(&CborValue::Integer(idx)) {
+            match map.get(&idx).cloned().map(ciborium::Value::into) {
```

Comment:
This shouldn't be necessary to clone and `Into::into`
```suggestion
            match map.get(&idx) {
```

Reply:
changed